chain/types/actor_event.go

```diff
@@ -0,0 +1,72 @@
+package types
+
+import (
+	"github.com/ipfs/go-cid"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-state-types/abi"
+)
+
+type ActorEventBlock struct {
+	// The value codec to match when filtering event values.
+	Codec uint64 `json:"codec"`
+
+	// The value to want to match on associated with the corresponding "event key"
+	// when filtering events.
+	// Should be a byte array encoded with the specified codec.
+	// Assumes base64 encoding when converting to/from JSON strings.
+	Value []byte `json:"value"`
+}
+
+type SubActorEventFilter struct {
+	Filter ActorEventFilter `json:"filter"`
+
+	// If true, all available matching historical events will be written to the response stream
+	// before any new real-time events that match the given filter are written.
+	// If `Prefill` is true and `FromEpoch` is set to latest, the pre-fill operation will become a no-op.
+	// if `Prefill` is false and `FromEpoch` is set to earliest, historical events will still be sent to the client.
+	Prefill bool `json:"prefill"`
```

Comment:
TODO: remove. Prefill is implied if `FromEpoch` (now `FromHeight`) is non-nil.

Reply:
@Stebalien I have a minor mismatch from our discussion with `Prefill`, but I think maybe I'm remembering it wrong since we did a couple of loops. We came to the conclusion that the use-case for "earliest you have" isn't clear so we're doing away with that (eventually to error if you specify a height we don't actually have). Without `Prefill` on the subscribe then I think we can collapse `SubActorEventFilter` and do away with it entirely so the same `ActorEventFilter` is provided to both APIs. It does mean that you must use `FromEpoch` as a signal to decide whether to fish around in historical events or not. The only differences between the two APIs is (1) the `chan` and (2) the ability to reach into the future for _both_ `Height` parameters for `Subscribe` whereas they are both maxxed at the current heaviest for `Get`.

Reply:
> I think we can collapse SubActorEventFilter and do away with it entirely so the same ActorEventFilter is provided to both APIs.

Yes.

> It does mean that you must use FromEpoch as a signal to decide whether to fish around in historical events or not

Yes.

> The only differences between the two APIs is

That and, for "get" `FromEpoch` and `ToEpoch` default to "latest". For "subscribe", they default to "next" (i.e., the epoch after the current epoch. Subtle difference that nobody's going to care about but still technically a difference.